lib/src/mistralai_client_dart_base.dart

```diff
@@ -103,6 +103,54 @@ class MistralAIClient {
     }
   }
 
+  Stream<ChatCompletionChunk> streamChat(ChatParams params) async* {
+    try {
+      final headers = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'Authorization': 'Bearer $apiKey',
+      };
+
+      // TODO(lgawron): endpoint building should be moved to separate class
+      // cleanup in https://github.com/nomtek/mistralai_client_dart/issues/13
+      final url = Uri.parse('$baseUrl/v1/chat/completions');
+
+      // TODO(lgawron): to add tests for this method we need to mock http client
+      // cleanup in https://github.com/nomtek/mistralai_client_dart/issues/13
+      final request = http.Request('POST', url)
+        ..headers.addAll(headers)
+        ..body = jsonEncode(
+          _mapChatParamsToRequestParams(params, stream: true),
+        );
+      // use send instead of post to be able to read stream response
+      final response = await _client.send(request).timeout(timeout);
+
+      var buffer = '';
+      final responseStream = response.stream.transform(utf8.decoder);
+      const dataPrefix = 'data:';
+      await for (final chunk in responseStream) {
+        // ignore: use_string_buffers
```

Comment:
Why are we ignoring this warning? Why don't use a string buffer / string builder right from the start? 🤔

Reply:
StringBuffer was not better here and more readable. I change the code in this commit to be more readable https://github.com/nomtek/mistralai_client_dart/pull/17/commits/e1683e613eaf15106c4f6b64abc8f820e7a9f120